staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -42,8 +42,14 @@ import (
 	utiltrace "k8s.io/apiserver/pkg/util/trace"
 )
 
-// PatchResource returns a function that will handle a resource patch
-// TODO: Eventually PatchResource should just use GuaranteedUpdate and this routine should be a bit cleaner
+// PatchResource returns a function that will handle a resource patch.
+//
+// Note that most k8s api operations treat RV as a precondition. PATCH has an
+// additional optimization: if RV differs but we can predict an equivalent
```

Comment:
if the client wants this behavior, they have to omit RV from their patch... if they include RV, that goes to etcd every time and produces an etcd-level conflict.

Reply:
Yes, it seems that patch always runs a bunch of logic that only makes sense if RV was set, but if you set RV the retries will always get a conflict. I have edited the comment and send @bgrant0607 an email asking to check my memory.